internal/harness/k3s/k3s.go

```diff
@@ -263,12 +263,24 @@ func (h *k3s) startSandbox(ctx context.Context, cli *docker.Client, resp *docker
 		return fmt.Errorf("getting kubeconfig: %w", err)
 	}
 
-	// Attach the sandbox to all networks the k3s service is also part of
+	networks := make(map[string]struct{})
+	for _, nw := range h.Sandbox.Networks {
+		networks[nw.ID] = struct{}{}
+	}
+
+	// Attach the sandbox to any networks k3s is also a part of, excluding any
+	// invalid networks or networks already attached (the daemon cannot deconflict
+	// these)
 	for nn, nw := range resp.NetworkSettings.Networks {
-		h.Sandbox.Networks = append(h.Sandbox.Networks, docker.NetworkAttachment{
-			Name: nn,
-			ID:   nw.NetworkID,
-		})
+		if nn == "" {
```

Comment:
How do we end up with unnamed networks?

Reply:
IIUC since we are building the networks map from the TF HCL definition for the sandbox, we shouldn't end up with network IDs that are `""`, so we should be able to rely on the `!ok` check below, right?